systemtests/tests/py3plug-fd-basic/python-modules/cancel-check-module.py

```diff
@@ -0,0 +1,189 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# BAREOS - Backup Archiving REcovery Open Sourced
+#
+# Copyright (C) 2023-2023 Bareos GmbH & Co. KG
+#
+# This program is Free Software; you can redistribute it and/or
+# modify it under the terms of version three of the GNU Affero General Public
+# License as published by the Free Software Foundation, which is
+# listed in the file LICENSE.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.
+#
+
+# this test module will emit a job message in every single callback there is
+# to make sure emitting a message won't break anything
+
+# import all the wrapper functions in our module scope
+from BareosFdWrapper import *
+
+from bareosfd import (
+    bRC_OK,
+    JobMessage,
+    DebugMessage,
+    M_INFO,
+    M_WARNING,
+    bFileType,
+    StatPacket,
+    FT_REG,
+    GetValue,
+    bVarJobId,
+    bEventCancelCommand,
+    RegisterEvents,
+)
+
+from BareosFdPluginBaseclass import BareosFdPluginBaseclass
+
+import sys
+from stat import S_IFREG, S_IFDIR, S_IRWXU
+from subprocess import Popen, PIPE
+import time
+
+
+@BareosPlugin
+class TestPlugin(BareosFdPluginBaseclass):
+    def __init__(self, plugindef):
+        super().__init__(plugindef)
+        JobMessage(M_INFO, "__init__('{}')\n".format(plugindef))
+
+        RegisterEvents([bEventCancelCommand])
+        self.files_to_backup = ["result"]
+        self.should_wait = True
+
+    def parse_plugin_definition(self, plugindef):
+        JobMessage(M_INFO, "parse_plugin_definition('{}')\n".format(plugindef))
+        ret = super().parse_plugin_definition(plugindef)
+        self.plugindef = plugindef
+        wait_opt = self.options["should_wait"]
+        if wait_opt != None:
```

Comment:
```suggestion
        if wait_opt is not None:
```
While this works in that context, it doesn't do what you'd expect. The proper way to check this is `if wait_opt is not None` which will actually check that `wait_opt` is not the `None` object.
Just doing `wait_opt != None` would presumably call `wait_opt._eq_(None)` which - depending on the type - may not behave well with `None` as a parameter.